compiler/pash_runtime.sh

```diff
@@ -141,7 +141,7 @@ else
         pash_redir_output echo "$$: (5) BaSh script exited with ec: $pash_runtime_final_status"
     else 
         function run_parallel() {
-            trap inform_daemon_exit SIGTERM SIGINT EXIT
+            trap inform_daemon_exit SIGTERM SIGINT
             export SCRIPT_TO_EXECUTE="$pash_script_to_execute"
             source "$RUNTIME_DIR/pash_restore_state_and_execute.sh"
             inform_daemon_exit
```

Comment:
Maybe the best solution would be to leave the trap as is, and remove the inform at the end of the call to run_parallel? This would make sure that we only call `inform_daemon_exit` once (I wonder whether the current implementation allows a subtle corner case now where SIGTERM is received while we are in inform). 

Based on your explanation (EXIT not being called on Ubuntu 20) my proposed fix shouldn't work on Ubuntu 20, so if we agree that this would be a better fix but it doesn't work on Ubuntu 20, let's leave a comment in the code to remember that we have it at the end and at the trap because for some reason the trap is not properly triggered, but if it was, that is how we would like to implement it

Reply:
That’d be the best solution, but it doesn’t work, unfortunately. State of the repo after the first failed CI run: https://github.com/binpash/pash/pull/739/files/509fcf00f62edd210d969708dfee0d26e4088919

I’ll add a comment mentioning this. I can add a quick check in inform_daemon_exit as for whether exit it’s been called already to avoid the sigterm corner-case, which’d avoid a future bug like this.

Reply:
A comment should be fine :) No need to solve it since it wasn't solved previously either!